src/Cardano/Wallet/Primitive/AddressDerivation.hs

```diff
@@ -24,7 +24,7 @@ module Cardano.Wallet.Primitive.AddressDerivation
     (
     -- * Polymorphic / General Purpose Types
     -- $use
-      Key
+      Key (..)
```

Comment:
Either had to do this, or move `signRaw` to AddressDerivation.hs

Reply:
I'd suggest to have `sign` part of the address derivation module, as:

```hs
sign
    :: SignTag   
    -> (Key 'AddrK XPrv, Passphrase "encryption") 
    -> Hash any
    -> Hash "signature" 
sign tag (Key k, pwd) (Hash payload) = CC.unXSignature $ CC.sign pwd k (tag <> payload)
```

Reply:
the `Signing` module only contains `mkStdTx` then 🤔 

_won't look more into this now_

Instead of passing in a `SignTag`, could do with `Hash "payload"`
```haskell
sign
    :: Hash "payload"
    -> (Key 'AddressK XPrv, Passphrase "encryption")
    -> Hash "signature"
sign (Hash payload) (Key k, (Passphrase pwd)) = Hash . CC.unXSignature $ CC.sign pwd k payload
```

and then everything but `sign` can live in the `Signing`-module… 🤔

Reply:
The problem was to extract the key from the `Key ... XPrv`, right ? It feels wrong to have sign in the `AddressDerivation` module in the end :| ... 
Perhaps, simply providing accessors to the underlying key `getKey`; and remove / relocate as well the encodeXPub  as part of the signing module?

Reply:
> getKey

I think that is decent.


Though the design/organization here / in general is not crystal-clear to me.

This is a Cardano.Wallet.**Primitive**.Signing module that depends on Cardano.Wallet.Binary. _Although both `...Primitive.Model` and `...Primitive.AddressDerivation` also depends on `Binary`._

Also would think it would make sense to have `Key` be opaque and the module it lives in be the only point of contact with `cardano-crypto`.

Here, in the signing code, there is also some ugly friction caused by conversion between `cardano-crypto` wrappers, `ByteArrayAccess`, `ByteString` and `Hash`. I wonder if something more holistic would be possible. Or we will never be able to escape it.

So `getKey` seems decent.

Reply:
Done.